docs/_guide/publish.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: guide
+title: Publish an element
+slug: publish
+---
+
+{::options toc_levels="1..3" /}
+* ToC
+{:toc}
+
+This page describes how to publish a LitElement component to npm.
+
+We recommend publishing JavaScript modules in standard ES2017. If you're writing your element in standard ES2017, you don't need to transpile for publication. If you're using TypeScript, or ES2017+ features such as decorators or class fields, you will need to transpile your element for publication.
+
+## Publishing to npm
+
+To publish your component to npm, [see the instructions on contributing npm packages](https://docs.npmjs.com/packages-and-modules/contributing-packages-to-the-registry).
+
+Your package.json configuration should have both the `main` and `module` fields:
+
+**package.json**
+
+```json
+{
+  "main": "my-element.js",
+  "module": "my-element.js"
+}
+```
+
+You should also create a README describing how to consume your component. A basic guide to consuming LitElement components is documented at [Use a component](use).
+
+## Transpiling with TypeScript
+
+When compiling your code from TypeScript to JavaScript, we recommend targeting ES2017 with Node.js module resolution. See the examples below for suggested options in tsconfig.json, and see also the [tsconfig.json documentation](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html) for more information.
+
+### Targeting ES2017 (Recommended)
+
+The following JSON sample is a partial tsconfig.json that uses recommended options for targeting ES2017:
+
+```json
+  "compilerOptions": {
+    "target": "ES2017",
+    "module": "ES2017",
+    "moduleResolution": "node",
+    "lib": ["ES2017", "DOM"],
+    "experimentalDecorators": true,
+    "outDir": "path/to/your-output-dir"
+  }
+```
+
+### Targeting ES5
+
+The following JSON sample is a partial tsconfig.json for those who need to target ES5:
+
+```json
+  "compilerOptions": {
+    "downlevelIteration": true,
+    "target": "ES5",
+    "module": "ES2015",
+    "moduleResolution": "node",
+    "lib": ["ES5", "DOM", "ScriptHost" ],
+    "experimentalDecorators": true,
+    "outDir": "path/to/your-output-dir"
+  }
+```
+
+<div class="alert alert-info">
+
+**You may need the compiler option `downlevelIteration` when targeting ES5**. The `downlevelIteration` option enables full support for [generators and the Iterator protocol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators#Generator_functions) when compiling to ES5. You need this option if your own code iterates over, for example, a `Set` or `Map`. 
```

Comment:
Since this is only if devs are using iteration themselves, it's generic TS advice that might not belong here. We made LitElement itself not do iteration on Maps and Sets anymore.

OTOH, lit-html uses iteration over user-provided data, so in the Babel section it might be important to say to use spec-compliant for/of support because presumably and build that compiles lit-html and LitElement will be using Babel (via Rollup/WebPack) to compile dependencies.

Reply:
Took it out, not to do with publishing

Reply:
@justinfagnani Staged again at https://20190128t182135-dot-polymer-lit-element.appspot.com